munch/__init__.py

```diff
@@ -197,12 +197,23 @@ def __repr__(self):
             >>> eval(repr(b))
             Munch(foo=Munch(lol=True), hello=42, ponies='are pretty!')
 
+            >>> with_spaces = Munch({'a b': 9, 'c': Munch({'simple': 5})})
+            >>> print (repr(with_spaces))
+            Munch({'a b': 9, 'c': Munch(simple=5)})
+            >>> eval(repr(with_spaces))
+            Munch({'a b': 9, 'c': Munch(simple=5)})
+
             (*) Invertible so long as collection contents are each repr-invertible.
         """
         keys = list(iterkeys(self))
         keys.sort()
-        args = ', '.join(['%s=%r' % (key, self[key]) for key in keys])
-        return '%s(%s)' % (self.__class__.__name__, args)
+        if any(" " in k for k in keys):
+            dict_content = ', '.join(["'%s': %r" % (key, self[key]) for key in keys])
```

Comment:
This still has issues with keys containing quotes etc. `%s` is bound to fail here, and I think `%r` is better...